recipes/opencv/4.x/conanfile.py

```diff
@@ -99,8 +111,11 @@ def configure(self):
         if not self.options.with_cuda:
             del self.options.with_cublas
             del self.options.with_cufft
-        self.options["libtiff"].jpeg = self.options.with_jpeg
-        self.options["jasper"].with_libjpeg = self.options.with_jpeg
+        if bool(self.options.with_jpeg):
+            if self.options.get_safe("with_jpeg2000") == "jasper":
+                self.options["jasper"].with_libjpeg = self.options.with_jpeg
+            if self.options.get_safe("with_tiff"):
+                self.options["libtiff"].jpeg = self.options.with_jpeg
```

Comment:
previous logic here was broken. We want to use the same jpeg backend in all dependencies, not to disable jpeg support in dependencies when jpeg is disabled in opencv.

Reply:
I tipically prefer to raise if values mismatch than to force a value for a dependency, but both approaches will work almost the same. 

Anyway, the `provides` mechanism should prevent mixing different `jpeg` implementations in the same graph (for the good and the bad), if recipes are declaring it.

Reply:
I understand, I didn't introduce this mechanism in this PR, it was already there but potentially broken.

Reply:
yes, eventually we will be able to remove this code altogether